src/PluginManagerAlter/Microsoft.Performance.Toolkit.PluginManager.Core.Alter.NuGet/NuGetPluginFetcher.cs

```diff
@@ -0,0 +1,102 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Performance.SDK;
+using Microsoft.Performance.Toolkit.PluginManager.Core.Alter.Transport;
+using NuGet.Configuration;
+using NuGet.Packaging.Core;
+using NuGet.Packaging;
+using NuGet.Protocol.Core.Types;
+using NuGet.Versioning;
+using NuGet.Protocol;
+using Microsoft.Performance.Toolkit.PluginManager.Core.Alter.Discovery;
+using System.Linq;
+using NuGet.Common;
+
+namespace Microsoft.Performance.Toolkit.PluginManager.Core.Alter.NuGet
+{
+    public sealed class NuGetPluginFetcher : IPluginFetcher
+    {
+        private static Guid NuGetId = Guid.Parse(PluginManagerConstants.NuGet);
+        
+        public Guid TypeId
+        {
+            get
+            {
+                return NuGetId;
+            }
+        }
+
+        public bool IsSupported(AvailablePlugin plugin)
```

Comment:
I wonder if this should return a `Task<bool>` and be async. This would allow checking against the upstream source to verify the plugin exists